x-pack/plugins/apm/server/routes/__test__/routeFailures.test.ts

```diff
@@ -13,31 +14,35 @@ import { initStatusApi } from '../status_check';
 import { initTracesApi } from '../traces';
 import { initTransactionsApi } from '../transactions';
 
-describe('route handlers fail properly', () => {
+describe('route handlers should fail with a Boom error', () => {
   let consoleErrorSpy: any;
 
   async function testRouteFailures(init: (server: Server) => void) {
     const mockServer = { route: jest.fn() };
     init((mockServer as unknown) as Server);
     expect(mockServer.route).toHaveBeenCalled();
 
-    const routes = mockServer.route.mock.calls;
+    const mockCluster = {
+      callWithRequest: () => Promise.reject(new Error('request failed'))
+    };
+    const mockConfig = { get: jest.fn() };
     const mockReq = {
       params: {},
       query: {},
-      pre: {
-        setup: {
-          config: { get: jest.fn() },
-          client: jest.fn(() => Promise.reject(new Error('request failed')))
+      server: {
+        config: () => mockConfig,
+        plugins: {
+          elasticsearch: {
+            getCluster: () => mockCluster
+          }
         }
       }
     };
 
+    const routes = flatten(mockServer.route.mock.calls);
     routes.forEach(async (route, i) => {
-      test(`route ${i + 1} of ${
-        routes.length
-      } should fail with a Boom error`, async () => {
-        await expect(route[0].handler(mockReq)).rejects.toMatchObject({
+      test(`${route.method} ${route.path}"`, async () => {
+        await expect(route.handler(mockReq)).rejects.toMatchObject({
```

Comment:
@jasonrhodes I made a minor change to the test output:

<img width="630" alt="screenshot 2018-11-27 at 13 37 33" src="https://user-images.githubusercontent.com/209966/49083024-a89b0200-f24b-11e8-81eb-44986ce6e32f.png">

ok?

Reply:
Nice!